app/bin/tools_backfill_packages.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart';
+import 'package:pub_dartlang_org/models.dart';
+
+import 'server_common.dart';
+import 'tools_common.dart';
+
+Future main(List<String> args) async {
+  int updatedPackages = 0;
+  await withProdServices(() async {
+    await fork(() async {
+      if (Platform.isMacOS) {
+        await initializeApiaryDatastore();
+      }
```

Comment:
I think it would be more readable to encapsulate the `Platform.isMacOS` specific code (as already mentioned somewhere).

Something like this, should do it:

```dart
Future withProdServices(Future fun()) {
  if (Platform.isMacOS) {
    return _withProdServices(() {
      return fork(() async {
        await initializeApiaryDatastore();
        return fun();
      });
    });
  } else {
    return _withProdServices(fun);
  }
}

Future _withProdServices(Future fun()) async {
  if (!Platform.environment.containsKey('GCLOUD_PROJECT') ||
      !Platform.environment.containsKey('GCLOUD_KEY')) {
    throw 'Missing GCLOUD_* environments for package:appengine';
  }
  return withAppEngineServices(() {
    registerUploadSigner(
        new ServiceAccountBasedUploadSigner(activeConfiguration.credentials));
    initBackend();
    return fun();
  });
```

Then everyone using `withProdServices()` doesn't need to care about it anymore